sympy/core/symbol.py

```diff
@@ -737,3 +739,50 @@ def traverse(symbols, frame):
         del frame  # break cyclic dependencies as stated in inspect docs
 
     return syms
+
+def disambiguate_symbols(iter):
+    """
+    This provides a method to disambiguate symbols that print the same.
+
+    Parameters
+    ==========
+
+    iter: Iterables.
+        List, Tuple or Set
+
+    Returns
+    =======
+
+    A sympy Tuple object containing disambiguate symbols.
+
+    Examples
+    ========
+
+    >>> from sympy.core.symbol import disambiguate_symbols
+    >>> from sympy import Tuple, Dummy, Symbol
+
+    >>> tup = Tuple(Symbol('_x'), Dummy('x'), Dummy('x'))
+    >>> disambiguate_symbols(tup)
+    (_x, _x_1, _x_2)
+
+    >>> disambiguate_symbols([Dummy('x'), Dummy('x')])
+    (_x, _x_1)
+
+    """
+    new_iter = iter
+    if not isinstance(new_iter, Basic):
+        new_iter=Tuple()
+        for i in iter:
+            new_iter += Tuple(i)
+    syms = ordered(new_iter.free_symbols, keys=new_iter.index)
```

Comment:
I see that you are thinking the arguments passed are symbols so you can just use the index to make the ordering canonical, but I was thinking that expressions would be passed and *internally* the symbols would be extracted (in which case index won't work and perhaps what I suggested in the `ok` function would be better.